Generate serialization of SharedMemoryHandle

donny-dont opened the pull request:
#### 08f49d42519ccb2d2e4725cd095609ddc6311383
<pre>
Generate serialization of SharedMemoryHandle
<a href="https://bugs.webkit.org/show_bug.cgi?id=258151">https://bugs.webkit.org/show_bug.cgi?id=258151</a>

Reviewed by Chris Dumez and Alex Christensen.

Split the generated serialization into two files, now platform specific
serializers are added to `WebKitPlatformGeneratedSerializers`.

Define serialization of `SharedMemoryHandle` in a `.serialization.in`
file. Mark serialization as `RValue` to ensure move semantics are used,
and `WebKitPlatform` so it ends up in the platform serializers.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/SharedMemory.cpp:
(IPC::ArgumentCoder&lt;WebKit::SharedMemoryHandle&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::SharedMemoryHandle&gt;::decode): Deleted.
* Source/WebKit/Platform/SharedMemory.h:
* Source/WebKit/Platform/SharedMemory.serialization.in: Added.
* Source/WebKit/Scripts/generate-serializers.py:
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Scripts/webkit/tests/WebKitPlatformGeneratedSerializers.cpp: Added.
(IPC::ArgumentCoder&lt;WebKit::PlatformClass&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::PlatformClass&gt;::decode):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Co-authored-by: Alex Christensen &lt;achristensen@apple.com&gt;
Canonical link: <a href="https://commits.webkit.org/265865@main">https://commits.webkit.org/265865@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b4f6457f97b7da4f5ca0f19ee0b3f17278b62c6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12072 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12418 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12720 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13818 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11651 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14834 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12432 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14348 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12236 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13047 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10222 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14233 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/12166 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10335 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18086 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11415 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11119 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14295 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11606 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9561 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10820 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2963 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15145 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11457 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Platform/SharedMemory.serialization.in:
```diff
@@ -0,0 +1,45 @@
+
+webkit_platform_headers: "SharedMemory.h" <wtf/ArgumentCoder.h> <wtf/RefCounted.h> "Encoder.h" "Decoder.h"
+#if USE(UNIX_DOMAIN_SOCKETS)
+webkit_platform_header: <wtf/unix/UnixFileDescriptor.h>
```
Add "ArgumentCodersUnix.h"

achristensen07 commented on Source/WebKit/Platform/SharedMemory.serialization.in:
```diff
@@ -0,0 +1,45 @@
+webkit_platform_header: <wtf/MachSendRight.h> "ArgumentCodersDarwin.h"
+#endif
+#if OS(WINDOWS)
+webkit_platform_header: <wtf/win/Win32Handle.h>
```
Add "ArgumentCodersWin.h"

donny-dont committed:
Generate serialization of SharedMemoryHandle
https://bugs.webkit.org/show_bug.cgi?id=258151

Reviewed by Chris Dumez and Alex Christensen.

Split the generated serialization into two files, now platform specific
serializers are added to `WebKitPlatformGeneratedSerializers`.

Define serialization of `SharedMemoryHandle` in a `.serialization.in`
file. Mark serialization as `RValue` to ensure move semantics are used,
and `WebKitPlatform` so it ends up in the platform serializers.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/SharedMemory.cpp:
(IPC::ArgumentCoder<WebKit::SharedMemoryHandle>::encode): Deleted.
(IPC::ArgumentCoder<WebKit::SharedMemoryHandle>::decode): Deleted.
* Source/WebKit/Platform/SharedMemory.h:
* Source/WebKit/Platform/SharedMemory.serialization.in: Added.
* Source/WebKit/Scripts/generate-serializers.py:
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Scripts/webkit/tests/WebKitPlatformGeneratedSerializers.cpp: Added.
(IPC::ArgumentCoder<WebKit::PlatformClass>::encode):
(IPC::ArgumentCoder<WebKit::PlatformClass>::decode):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Co-authored-by: Alex Christensen <achristensen@apple.com>
Canonical link: https://commits.webkit.org/265865@main
